### PR TITLE
Add .delete to ReflectionQuery CLOSES #602

### LINKED ISSUE #602: Support `delete` in `ReflectionQuery`
We should have a `@Delete` annotation, like the `@Select` we currently have, this will let us support delete with a where clause instead of using `hibernateSession.delete(entity);`

### PR DESCRIPTION
**Pull request by jjestrel**

### CODE REVIEW

File: misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt [hunk 1]
```diff
@@ -420,6 +420,42 @@ class ReflectionQueryFactoryTest {
...
+  @Test
+  fun delete() {
+    transacter.transaction { session ->
+      session.save(DbMovie("Jurassic Park", LocalDate.of(1993, 6, 9)))
+      session.save(DbMovie("Rocky", LocalDate.of(1976, 11, 21)))
+      session.save(DbMovie("Star Wars", LocalDate.of(1977, 5, 25)))
+    }
+
+    transacter.transaction { session ->
+      queryFactory.newQuery<OperatorsMovieQuery>()
```
Comment (alanpaulin) [inline misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt]: Can you `assertThat` the result of the delete is `2`?

Comment (jjestrel) [reply misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt]: Done

File: misk-hibernate/src/main/kotlin/misk/hibernate/Query.kt [hunk 2]
```diff
@@ -11,6 +11,8 @@ interface Query<T> {
 
   fun list(session: Session): List<T>
 
+  fun delete(session: Session): Int
```
Comment (wesleyk) [inline misk-hibernate/src/main/kotlin/misk/hibernate/Query.kt]: javadoc? (mention what is returned)

File: misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt [hunk 3]
```diff
@@ -420,6 +420,43 @@ class ReflectionQueryFactoryTest {
     }
   }
 
+  @Test
+  fun deleteFailsWithOrderBy() {
+    transacter.transaction { session ->
```
Comment (wesleyk) [inline misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt]: /nit is it worth even setting this data up?

File: misk-hibernate/src/main/kotlin/misk/hibernate/Query.kt (hunk 2 above)
Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/Query.kt]: Added

File: misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt (hunk 3 above)
Comment (jjestrel) [reply misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt]: I changed the test to make sure the delete didn't go through

**Pull request merged**